sitecore/9.0.1 rev. 171219 XM1/VS2017/Website/Website/Website.csproj

```diff
@@ -0,0 +1,170 @@
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=1.0.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.1.0.7\lib\net45\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.CSharp" />
+    <Reference Include="Sitecore.Kernel, Version=11.1.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\Sitecore.Kernel.NoReferences.9.0.171219\lib\NET462\Sitecore.Kernel.dll</HintPath>
+    </Reference>
+    <Reference Include="Sitecore.Mvc, Version=2.1.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <HintPath>..\packages\Sitecore.Mvc.NoReferences.9.0.171219\lib\NET462\Sitecore.Mvc.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Data.DataSetExtensions" />
+    <Reference Include="System.Web.DynamicData" />
+    <Reference Include="System.Web.Entity" />
+    <Reference Include="System.Web.ApplicationServices" />
+    <Reference Include="System.ComponentModel.DataAnnotations" />
+    <Reference Include="System" />
+    <Reference Include="System.Data" />
+    <Reference Include="System.Web.Extensions" />
+    <Reference Include="System.Drawing" />
+    <Reference Include="System.Web" />
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Configuration" />
+    <Reference Include="System.Web.Services" />
+    <Reference Include="System.EnterpriseServices" />
+    <Reference Include="System.Xml.Linq" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="System.Web.Razor">
+      <HintPath>..\packages\Microsoft.AspNet.Razor.3.2.3\lib\net45\System.Web.Razor.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Web.Webpages">
+      <HintPath>..\packages\Microsoft.AspNet.Webpages.3.2.3\lib\net45\System.Web.Webpages.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Web.Webpages.Deployment">
+      <HintPath>..\packages\Microsoft.AspNet.Webpages.3.2.3\lib\net45\System.Web.Webpages.Deployment.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Web.Webpages.Razor">
+      <HintPath>..\packages\Microsoft.AspNet.Webpages.3.2.3\lib\net45\System.Web.Webpages.Razor.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Web.Helpers">
+      <HintPath>..\packages\Microsoft.AspNet.Webpages.3.2.3\lib\net45\System.Web.Helpers.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Web.Infrastructure">
+      <HintPath>..\packages\Microsoft.Web.Infrastructure.1.0.0.0\lib\net40\Microsoft.Web.Infrastructure.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Web.Mvc">
+      <HintPath>..\packages\Microsoft.AspNet.Mvc.5.2.3\lib\net45\System.Web.Mvc.dll</HintPath>
+    </Reference>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="Global.asax" />
+    <Content Include="Web.config" />
```

Comment:
I think having Web.config as content will overwrite Sitecore's web.config. Should mark this as "None" and generate transform files if necessary.

Reply:
Not sure whether I understand the requirement. This sample is showcasing a Sitecore concrete project. All projects (Sitecore sites), as far as I know, need to modify their Web.config file. A typical example is using <customErrors mode="Off" /> while developing and something else in a production release.

It is also typical to use standard VS transform files such as "Web.Debug.config" for those transformations, as it is done in this project (setting the "Web.config" file as content). With this settings, the Web.config in the container is overwritten with the transformed version generated by the project.

Can you please elaborate a bit more about what change should be done?

Reply:
I was just thinking of the "Habitat way" where you don't deploy the Web.config file but apply transforms using gulp. Helps during upgrades and prevents overwriting of Sitecore-specific web.config items. I'm not sure what it would mean when deploying to containers.

If you look at [Habitat Website Project](https://github.com/Sitecore/Habitat/blob/master/src/Project/Habitat/code/Sitecore.Habitat.Website.csproj)  the web project has the web.config set to None.

  `<None Include="web.config" />
    <None Include="web.Debug.config">
      <DependentUpon>web.config</DependentUpon>
    </None>
    <None Include="web.Release.config">
      <DependentUpon>web.config</DependentUpon>
    </None>`

Then in the Habitat project, the [gulpfile](https://github.com/Sitecore/Habitat/blob/master/gulpfile.js) takes care of applying all of the transforms required by the various projects:

`gulp.task("04-Apply-Xml-Transform")`

Since this sample project is specific to one version of Sitecore it's probably okay to leave it like that, keep in mind that your web.config will overwrite the installed web.config and any Sitecore-specific items will be lost if not also included in your project's web.config.

Reply:
I see your point now. It is actually a good idea.

This sample is meant to be simple. I wanted to use the bare minimum infrastructure to integrate Docker in VS2017 which is the main development environment for Sitecore at the moment (at least the one we use in ClearPeople, where I work). The intention is showcasing the bare minimum required to run Sitecore in a container.

I am thinking of a new demo including a TDS sample project and gulp with some automation such as automatically updating the "host" file in de development environment in order to access the Sitecore site through its URL (instead of the IP). Once we have gulp, the web.config transform can be easily included.

Other samples may include installation of modules such as SXA or EXM, or other proposals from the community that we find interesting.

May I suggest that these ideas are added to the list of "Issues/enhancements"?

Anyway, the progress will depend on time availability, which is not guaranteed, so let's progress bit by bit.